homeassistant/components/mqtt/sensor.py

```diff
@@ -53,18 +53,48 @@
 
 DEFAULT_NAME = "MQTT Sensor"
 DEFAULT_FORCE_UPDATE = False
-PLATFORM_SCHEMA = mqtt.MQTT_RO_PLATFORM_SCHEMA.extend(
-    {
-        vol.Optional(CONF_DEVICE_CLASS): DEVICE_CLASSES_SCHEMA,
-        vol.Optional(CONF_EXPIRE_AFTER): cv.positive_int,
-        vol.Optional(CONF_FORCE_UPDATE, default=DEFAULT_FORCE_UPDATE): cv.boolean,
-        vol.Optional(CONF_LAST_RESET_TOPIC): mqtt.valid_subscribe_topic,
-        vol.Optional(CONF_LAST_RESET_VALUE_TEMPLATE): cv.template,
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-        vol.Optional(CONF_STATE_CLASS): STATE_CLASSES_SCHEMA,
-        vol.Optional(CONF_UNIT_OF_MEASUREMENT): cv.string,
-    }
-).extend(MQTT_ENTITY_COMMON_SCHEMA.schema)
+
+
+def validate_options(conf):
+    """Validate options.
+
+    If last reset topic is present it must be same as the state topic.
+    """
+    if (
+        CONF_LAST_RESET_TOPIC in conf
+        and CONF_STATE_TOPIC in conf
+        and conf[CONF_LAST_RESET_TOPIC] != conf[CONF_STATE_TOPIC]
+    ):
+        _LOGGER.warning(
+            "'%s' must be same as '%s'", CONF_LAST_RESET_TOPIC, CONF_STATE_TOPIC
+        )
+
+    if CONF_LAST_RESET_TOPIC in conf and CONF_LAST_RESET_VALUE_TEMPLATE not in conf:
+        _LOGGER.warning(
```

Comment:
Why is this a warning and not an error ?

Reply:
And can we use `vol.Inclusive` in that case ?

Reply:
The idea is to not enforce the changes in 2021.9 by rejecting the config, but instead just log a warning.
Then in 2021.10, reject the config.